blocks/product-comparison-table/product-comparison-table.js

```diff
@@ -0,0 +1,101 @@
+import getMockData from './product-mock-data.js';
+
+const fakeData = getMockData();
+
+function buildPriceContainer(productName, numberOfDevices, elementToReplace) {
+  const priceContainer = document.createElement('div');
+  priceContainer.classList.add('product-comparison-price');
+
+  const productData = fakeData
+    .filter((product) => product.data.product.product_name === productName);
+  if (productData.length === 0) {
+    return;
+  }
+  const variationPrice = productData[0].data.product.variations[numberOfDevices][1];
+  const productVariationPrice = variationPrice.price;
+  const productVariationDiscountPrice = variationPrice.discount.discounted_price;
+  const priceLabel = variationPrice.currency_label;
+
+  if (productVariationDiscountPrice) {
+    const oldPriceContainer = document.createElement('div');
+    oldPriceContainer.classList.add('old-price-container');
+    oldPriceContainer.innerHTML = `<p>Old Price <span class='price-strike-through'>${productVariationPrice} ${priceLabel}</span></p>`;
+    priceContainer.appendChild(oldPriceContainer);
+  }
+
+  const currentPriceContainer = document.createElement('div');
+  currentPriceContainer.classList.add('current-price-container');
+  currentPriceContainer.innerHTML = `<p>${productVariationDiscountPrice ?? productVariationPrice} ${priceLabel}</p>`;
+  priceContainer.appendChild(currentPriceContainer);
+
+  elementToReplace.replaceWith(priceContainer);
+}
+
+export default function decorate(block) {
```

Comment:
this is a rather long function - it might make sense to split into into smaller, more focused functions as it is rather hard to understand all the steps happening inside, like
- add accesibility roles
- mark header active
- decorate icons
- create headers

Reply:
updated